src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlCubeTranslator.cs

```diff
@@ -0,0 +1,307 @@
+        IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+    {
+        if (member.DeclaringType != typeof(NpgsqlCube))
+        {
+            return null;
+        }
+
+        return member.Name switch
+        {
+            nameof(NpgsqlCube.Dimensions)
+                => _sqlExpressionFactory.Function(
+                    "cube_dim",
+                    [instance!],
+                    nullable: true,
+                    argumentsPropagateNullability: TrueArrays[1],
+                    typeof(int)),
+
+            nameof(NpgsqlCube.IsPoint)
+                => _sqlExpressionFactory.Function(
+                    "cube_is_point",
+                    [instance!],
+                    nullable: true,
+                    argumentsPropagateNullability: TrueArrays[1],
+                    typeof(bool)),
+
+            nameof(NpgsqlCube.LowerLeft)
+                => throw new InvalidOperationException(
+                    $"The '{nameof(NpgsqlCube.LowerLeft)}' property cannot be translated to SQL. " +
+                    $"To access individual lower-left coordinates in queries, use indexer syntax (e.g., cube.LowerLeft[index]) instead."),
+
+            nameof(NpgsqlCube.UpperRight)
+                => throw new InvalidOperationException(
+                    $"The '{nameof(NpgsqlCube.UpperRight)}' property cannot be translated to SQL. " +
+                    $"To access individual upper-right coordinates in queries, use indexer syntax (e.g., cube.UpperRight[index]) instead."),
+
+            _ => null
+        };
+    }
+
+    private SqlExpression? TranslateSubset(IReadOnlyList<SqlExpression> arguments)
+    {
+        // arguments[0] is the cube
+        // arguments[1] is the int[] indexes array
+
+        // For subset, we need to convert the C# zero-based indexes to PostgreSQL one-based indexes
+        SqlExpression convertedIndexes;
+
+        switch (arguments[1])
+        {
+            case PgNewArrayExpression { Expressions: var expressions }
+                when expressions.All(e => e is SqlConstantExpression { Value: int }):
+                // OPTIMIZATION: For constant array literals (from params expansion), convert at translation time
+                var oneBasedConstants = expressions
+                    .Cast<SqlConstantExpression>()
+                    .Select(e => _sqlExpressionFactory.Constant((int)e.Value! + 1))
+                    .ToArray();
+
+                convertedIndexes = _sqlExpressionFactory.NewArray(oneBasedConstants, typeof(int[]));
+                break;
+
+            case SqlConstantExpression { Value: int[] constantArray }:
+                // OPTIMIZATION: For constant arrays, convert at translation time (no SQL overhead)
+                var oneBasedArray = constantArray
+                    .Select(i => _sqlExpressionFactory.Constant(i + 1))
+                    .ToArray();
+
+                convertedIndexes = _sqlExpressionFactory.NewArray(oneBasedArray, typeof(int[]));
```

Comment:
I think you should be creating a constant here, not a NewArray - just as the incoming constant array is constant, so should the outgoing.